src/library/Uno.Toolkit.Material/MaterialToolkitTheme.cs

```diff
@@ -0,0 +1,203 @@
+
+		#region FontOverrideDictionary (DP)
+		/// <summary>
+		/// (Optional) Gets or sets a <see cref="ResourceDictionary"/> containing overrides for the default Uno.Material <see cref="FontFamily"/> resources
+		/// </summary>
+		public ResourceDictionary FontOverrideDictionary
+		{
+			get => (ResourceDictionary)GetValue(FontOverrideDictionaryProperty);
+			set => SetValue(FontOverrideDictionaryProperty, value);
+		}
+
+		public static DependencyProperty FontOverrideDictionaryProperty { get; } =
+			DependencyProperty.Register(
+				nameof(FontOverrideDictionary),
+				typeof(ResourceDictionary),
+				typeof(MaterialToolkitTheme),
+				new PropertyMetadata(null, OnFontOverrideChanged));
+
+		private static void OnFontOverrideChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+		{
+			if (d is MaterialToolkitTheme { _isFontOverrideMuted: false } toolkitTheme)
+			{
+				toolkitTheme.UpdateSource();
+			}
+		}
+		#endregion
+
+		#region ColorOverrideDictionary (DP)
+		/// <summary>
+		/// (Optional) Gets or sets a <see cref="ResourceDictionary"/> containing overrides for the default Uno.Material <see cref="Color"/> resources
+		/// </summary>
+		/// <remarks>The overrides set here should be re-defining the <see cref="Color"/> resources used by Uno.Material, not the <see cref="SolidColorBrush"/> resources</remarks>
+		public ResourceDictionary ColorOverrideDictionary
+		{
+			get => (ResourceDictionary)GetValue(ColorOverrideDictionaryProperty);
+			set => SetValue(ColorOverrideDictionaryProperty, value);
+		}
+
+		public static DependencyProperty ColorOverrideDictionaryProperty { get; } =
+			DependencyProperty.Register(
+				nameof(ColorOverrideDictionary),
+				typeof(ResourceDictionary),
+				typeof(MaterialToolkitTheme),
+				new PropertyMetadata(null, OnColorOverrideChanged));
+
+		private static void OnColorOverrideChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+		{
+			if (d is MaterialToolkitTheme { _isColorOverrideMuted: false } toolkitTheme)
+			{
+				toolkitTheme.UpdateSource();
+			}
+		}
+		#endregion
+
+		private void SetColorOverrideSilently(ResourceDictionary colorOverride)
+		{
+			try
+			{
+				_isColorOverrideMuted = true;
+				ColorOverrideDictionary = colorOverride;
+			}
+			finally
+			{
+				_isColorOverrideMuted = false;
+			}
+		}
+
+		private void SetFontOverrideSilently(ResourceDictionary fontOverride)
+		{
+			try
+			{
+				_isFontOverrideMuted = true;
+				FontOverrideDictionary = fontOverride;
+			}
+			finally
+			{
+				_isFontOverrideMuted = false;
+			}
+		}
+
+		public MaterialToolkitTheme() : this(colorOverride: null, fontOverride: null)
+		{
+			
+		}
+
+		public MaterialToolkitTheme(ResourceDictionary? colorOverride = null, ResourceDictionary? fontOverride = null)
+		{
+			if (colorOverride is { })
+			{
+				SetColorOverrideSilently(colorOverride);
+			}
+
+			if (fontOverride is { })
+			{
+				SetFontOverrideSilently(fontOverride);
+			}
+
+			UpdateSource();
```

Comment:
```suggestion
			SetColorOverrideSilently(colorOverride);
			SetFontOverrideSilently(fontOverride);

			UpdateSource();
```
and change the setters signature to: 
```diff
-private void SetFontOverrideSilently(ResourceDictionary fontOverride)
+private void SetFontOverrideSilently(ResourceDictionary? fontOverride)
```